### DISCUSSION TITLE
Error in bind python function with C#'s delegate

### DISCUSSION THREAD

**Discussion by GentlerMan** [pythonnet/pythonnet#1685] [Q&A]
### Environment

-   Pythonnet version:2.5.2
-   Python version:3.8
-   Operating System:win10
-   .NET Runtime:4.7.2

### Details

    I want to bind python function with C#'s delegate method in winform

-   C# Code
```
public delegate string CallBackFunc(string content);
using (Py.GIL())
  {
      using (PyScope scope = Py.CreateScope())
      {
          scope.Set("main", this);
          scope.Exec(pycode);
      }
  }
```

-   Python Code
```
def test(input):
   return "OK"
res = main.CallBackFunc(test)
```

When I test in program ,GET the Error Infomation:Type was not found "System. Reflection. Emit. TypeBuilder" on the constructor,
What should I do For this?

**Comment by filmor** [comment]
1. This is not a support forum, it is not an adequate answer to me closing your ticket #1505 to just open a new one
2. Does this also happen with the current master branch?

**Comment by GentlerMan** [comment]
@filmor 
1.Thank you for attention, Because of the lack of relevant information about delegate, I have been beseted with this problem for several days, I just found some solutions ,such as [#1471](https://github.com/pythonnet/pythonnet/issues/1471),but it's no helpful for this.
2.I just test the newest release (Ver. 2.5.2), I am not sure whether  this also happen with other version.
